eogrow/cli.py

```diff
@@ -138,27 +130,38 @@ def ray(
         cli_variables: Tuple[str],
         test_patches: Tuple[int],
     ) -> None:
-        """Command for running an eo-grow pipeline on a remote Ray cluster of AWS EC2 instances
+        """Command for running an eo-grow pipeline on a remote Ray cluster of AWS EC2 instances. The provided config is
+        fully constructed and uploaded to the cluster head in the `~/.synced_configs/` directory, where it is then
+        executed. A custom suffix is added to distinguish runs which use the same config multiple times.
 
         \b
         Example:
             eogrow-ray cluster.yaml config_files/config.json
         """
+        if start_cluster:
+            subprocess.check_call(f"ray up -y {cluster_yaml}", shell=True)
+
         if stop_cluster and (use_screen or use_tmux):
             raise NotImplementedError("It is not clear how to combine stop flag with either screen or tmux flag")
 
-        configs = collect_configs_from_path(config_filename)
-        encoded_configs = encode_config_list(configs)
+        crude_configs = collect_configs_from_path(config_filename)
+        raw_configs = [interpret_config_from_dict(config) for config in crude_configs]
+        remote_path = get_cluster_config_path(config_filename)
+
+        with NamedTemporaryFile(mode="w", delete=True, suffix=".json") as local_path:
+            json.dump(raw_configs, local_path)
+            subprocess.check_call(f"ray rsync_up {cluster_yaml} {local_path.name!r} {remote_path!r}", shell=True)
```

Comment:
wasn't sure whether this should be a `check_call` since we tend to use `call` elsewhere. In the [docs](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module) it mentions that `run` should be used if possible.

Reply:
will change to `run`, since we don't do anything with the returned object anyway

Reply:
done